src/prefect/experimental/workers/base.py

```diff
@@ -36,6 +107,8 @@ def __bool__(self):
 @register_base_type
 class BaseWorker(abc.ABC):
     type: str
+    job_configuration: Optional[BaseJobConfiguration] = None
```

Comment:
I'm probably just not thinking about this clearly, but the `job_configuration` isn't actually optional for the a given child worker. But I don't want set the `job_configuration` on the BaseWorker. This is the equivalent of an abstract attribute. If there's a better way to designate this, I'm happy to change it.

Reply:
Remove `Optional` and don't set a value